lib/miq_expression/field.rb

```diff
@@ -18,6 +18,12 @@ def self.parse(field)
         parsed_params[:virtual_custom_column])
   end
 
+  def self.get_col_name(field)
+    return field.split('.').last.tr('-', '.') unless is_field?(field)
+    parsed_field = parse(field)
+    parsed_field.associations.push(parsed_field.column).join('.')
```

Comment:
Using `.push` here *I think* will corrupt the `associations` data on the field object. That's OK here because the object is short-lived but might lead to some confusion if this changes....the non-destructive `+ [other]` might be better?